From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Wed, 20 Jul 2022 11:31:02 +0200
Subject: [PATCH] Revert "printk: Skip console drivers on PREEMPT_RT."
Origin: https://www.kernel.org/pub/linux/kernel/projects/rt/5.19/older/patches-5.19-rc8-rt9.tar.xz

Revert the previous change and allow printing on consoles now that the atomic
consoles and the printing thread is available.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 kernel/printk/printk.c |   10 ----------
 1 file changed, 10 deletions(-)

--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -3298,16 +3298,6 @@ void console_unlock(void)
 	}
 
 	/*
-	 * On PREEMPT_RT it is not possible to invoke console drivers with
-	 * disabled interrupts and or preemption. Therefore all drivers are
-	 * skipped and the output can be retrieved from the buffer.
-	 */
-	if (IS_ENABLED(CONFIG_PREEMPT_RT)) {
-		__console_unlock();
-		return;
-	}
-
-	/*
 	 * Console drivers are called with interrupts disabled, so
 	 * @console_may_schedule should be cleared before; however, we may
 	 * end up dumping a lot of lines, for example, if called from
